pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -98,26 +105,28 @@ def clip_gradients(self, optimizer: Optimizer, clip_val: Union[int, float], norm
             return
 
         parameters = list(self.master_params(optimizer))
-
-        max_norm = grad_clip_val
-
-        if isinstance(parameters, torch.Tensor):
-            parameters = [parameters]
-        parameters = list(filter(lambda p: p.grad is not None, parameters))
-
-        device = parameters[0].device
-
-        if norm_type == math.inf:
-            total_norm = max(p.grad.data.abs().max() for p in parameters)
-        else:
-            out = torch.empty(len(parameters), device=device)
-            for i, p in enumerate(parameters):
-                torch.norm(p.grad.data.to(device), norm_type, out=out[i])
-            total_norm = torch.norm(out, norm_type)
-
-        eps = self.EPSILON
-
-        clip_coef = torch.tensor(max_norm, device=device) / (total_norm + eps)
-        clip_coef = torch.min(clip_coef, torch.ones_like(clip_coef))
-        for p in parameters:
-            p.grad.data.mul_(clip_coef.to(p.grad.data.device))
+        if gradient_clip_algorithm == GradClipAlgorithmType.VALUE:
+            torch.nn.utils.clip_grad_value_(parameters, clip_value=grad_clip_val)
+        elif gradient_clip_algorithm == GradClipAlgorithmType.NORM:
+            max_norm = grad_clip_val
```

Comment:
with the `if/else`, it's very easy to miss support in a plugin should a new `gradient_clip_algorithm` value be added.

this might be overkill, but should each precision plugin maintain a `Dict[GradClipAlgorithmType, Callable]` ? at least that will fail loudly should we try looking up that `gradient_clip_algorithm` in the dict

Reply:
So, you mean, managing `Dict[GradClipAlgorithmType, Callable]` as an attribute of every `PrecisionPlugin`, and making two callable function, `clip_grad_by_value`  and `clip_grad_by_norm` would be better.

I agree with you. Please check what I told you is the same as your comment. I'll be work in after your confirmation.

Reply:
Any new opinions from other reviewers?

Reply:
@dhkim0225 correct that's what i meant